Assets/VRTK/Scripts/Locomotion/VRTK_SlingshotJump.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Slingshot Jump|Locomotion|20120
+namespace VRTK
+{
+    using System;
+    using GrabAttachMechanics;
+    using UnityEngine;
+
+    /// <summary>
+    /// Event Payload
+    /// </summary>
+    /// <param name="target">The GameObject of the interactable object that is being interacted with by the controller.</param>
+    public struct SlingshotJumpEventArgs
+    {
+        public GameObject target;
+    }
+
+    /// <summary>
+    /// Event Payload
+    /// </summary>
+    /// <param name="sender">this object</param>
+    /// <param name="e"><see cref="PlayerClimbEventArgs"/></param>
+    public delegate void SlingshotJumpEventHandler(object sender, SlingshotJumpEventArgs e);
+
+    /// <summary>
+    /// Slingshot Jump allows player jumping based on the direction and amount pulled back of each controller. This slingshots the player in defined direction and speed.
+    /// </summary>
+    /// <example>
+    /// `VRTK/Examples/037_CameraRig_ClimbingFalling` shows how to set up a scene with slingshot jumping. This script just needs to be added to the PlayArea object and the requested forces and buttons set.
+    /// </example>
+    [AddComponentMenu("VRTK/Scripts/Locomotion/VRTK_SlingshotJump")]
+    [RequireComponent(typeof(VRTK_BodyPhysics))]
```

Comment:
My one concern here is that this will force the script to be on the same GameObject as the BodyPhysics script.

Other scripts have now moved away from using `RequireComponent` and have a public parameter (usually under a header of `Custom Settings`) that allows you to inject the required script into the component.

E.g. a param called `bodyPhysics` and if you leave it blank then it will attempt to look for the BodyPhysics script on Component or Parent. If it can't find a body physics script after all that then it's fine to throw an error (and make sure all references to the body physics script are null checked).